### PR TITLE
Add base contracts for LeverageEngine and dependencies

### PR DESCRIPTION
**Pull request by CoreyMartin**
In person review @yotamdaniel

### CODE REVIEW

File: contracts/LeverageAllocator.sol [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import "hardhat/console.sol";
+
+/// @title LeverageEngine is the user facing contract that initiates a leveraged position and returns an NFT to the user's wallet
```
Comment (CoreyMartin) [inline contracts/LeverageAllocator.sol:6]: fix

File: test/ContractTestContext.test.js [hunk 2]
```diff
@@ -37,4 +38,20 @@ describe("ContractTestContext", function () {
             });
         });
     });
+
+    describe("Basic contract import functions", async function () {
+        it("Should should be defined if contract exists", async function () {
+            // String in equal is 2^256 -1, max uint256
+            await expect(contractTestContext.coordinator).to.not.be.undefined;
+        });
+        it("Should should be undefined if contract does not exist", async function () {
```
Comment (jasonstockman) [inline test/ContractTestContext.test.js:47]: typo

Comment (jasonstockman) [inline test/ContractTestContext.test.js:47]:
```suggestion
        it("Should be undefined if contract does not exist", async function () {
```

Comment (jasonstockman) [inline test/ContractTestContext.test.js:47]:
```suggestion
        it("Should be undefined if contract does not exist", async function () {
```

**Review by jasonstockman — APPROVED** [review]

**Pull request merged**